File: go/src/leetcode/editor/cn/0252_meeting_rooms.go
package main

import "sort"

func main() {
	println("252、meeting_rooms")
}

// leetcode submit region begin(Prohibit modification and deletion)
func canAttendMeetings(intervals [][]int) bool {
	sort.Slice(intervals, func(i, j int) bool {
		return intervals[i][0] < intervals[j][0]
	})
	for i := 0; i < len(intervals)-1; i++ {
		if intervals[i][1] > intervals[i+1][0] {
			return false
		}
	}
	return true
}

//leetcode submit region end(Prohibit modification and deletion)

File: rust/src/leetcode/editor/cn/_0252_meeting_rooms.rs
#[cfg(test)]
mod tests {
    #[allow(unused_imports)]
    use super::*;

    #[test]
    fn test() {
        println!("252、meeting_rooms");
    }
}

#[allow(dead_code)]
struct Solution;

//leetcode submit region begin(Prohibit modification and deletion)
impl Solution {
    #[allow(dead_code)]
    pub fn can_attend_meetings(mut intervals: Vec<Vec<i32>>) -> bool {
        intervals.sort_unstable_by(|x, y| x[0].cmp(&y[0]));
        for i in 1..intervals.len() {
            if intervals[(i - 1) as usize][1] > intervals[i as usize][0] {
                return false;
            }
        }
        true
    }

    #[allow(dead_code)]
    pub fn can_attend_meetings0(mut intervals: Vec<Vec<i32>>) -> bool {
        intervals.sort_unstable_by(|x, y| x[0].cmp(&y[0]));
        intervals.windows(2).all(|x| x[0][1] <= x[1][0])
    }
}
//leetcode submit region end(Prohibit modification and deletion)
